### PR TITLE
Use OPA for RBAC

### PR DESCRIPTION
**Pull request by emosbaugh**

### CODE REVIEW

File: pkg/rbac/rego.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+import (
+	"context"
+
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/rego"
+	"github.com/pkg/errors"
+)
+
+var regoModule string = `
```
Comment (salahalsaleh) [inline pkg/rbac/rego.go:11]: would be nice to have this in a separate .rego file

Comment (emosbaugh) [reply pkg/rbac/rego.go:11]: then we will have to use gobindata. we can do that later if necessary

File: kotsadm/pkg/handlers/app.go [hunk 2]
```diff
@@ -328,6 +328,8 @@ type RemoveAppResponse struct {
 }
 
 func RemoveApp(w http.ResponseWriter, r *http.Request) {
+	fmt.Println("HERE")
```
Comment (emosbaugh) [inline kotsadm/pkg/handlers/app.go:331]: fix this

File: kotsadm/pkg/handlers/middleware.go [hunk 3]
```diff
@@ -23,6 +24,7 @@ func CorsMiddleware(next http.Handler) http.Handler {
 func RequireValidSessionMiddleware(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		sess, err := requireValidSession(w, r)
+		fmt.Println("requireValidSession", sess.ID, err)
```
Comment (salahalsaleh) [inline kotsadm/pkg/handlers/middleware.go:27]: clean this

File: kotsadm/pkg/gitops/gitops.go [hunk 4]
```diff
@@ -627,21 +627,28 @@ func CreateGitOpsCommit(gitOpsConfig *GitOpsConfig, appSlug string, appName stri
 		return "", err
 	}
 
-	filePath := filepath.Join(workDir, gitOpsConfig.Path, fmt.Sprintf("%s.yaml", appSlug))
+	dirPath := filepath.Join(workDir, gitOpsConfig.Path)
```
Comment (emosbaugh) [inline kotsadm/pkg/gitops/gitops.go:630]: revert this

File: kotsadm/pkg/handlers/middleware.go (hunk 3 above)
Comment (emosbaugh) [inline kotsadm/pkg/handlers/middleware.go:27]: remove this

File: kotsadm/pkg/policy/policy.go [hunk 5]
```diff
@@ -35,32 +35,33 @@ func Must(p *Policy, err error) *Policy {
...
+			return action, resource, "", err
 		}
 		for key, val := range additionalVars {
 			vars[key] = val
 		}
 	}
 	var buf bytes.Buffer
 	err = p.resourceTemplate.Execute(&buf, vars)
-	return p.action, buf.String(), err
+	appSlug, _ = vars["appSlug"]
```
Comment (salahalsaleh) [inline kotsadm/pkg/policy/policy.go:65]: won't this always overwrite the appSlug?

File: pkg/rbac/rego.go [hunk 6]
```diff
@@ -53,13 +53,23 @@ func init() {
 	}
 }
 
-func CheckAccess(ctx context.Context, action, resource string, roles []string) (bool, error) {
+func CheckAccess(ctx context.Context, action, resource, appSlug string, roles []string) (bool, error) {
+	allowRolePolicies := DefaultAllowRolePolicies
+	denyRolePolicies := DefaultDenyRolePolicies
+	if appSlug != "" {
```
Comment (salahalsaleh) [inline pkg/rbac/rego.go:59]: should this also include the support role?

File: kotsadm/pkg/policy/policy.go (hunk 5 above)
Comment (emosbaugh) [reply kotsadm/pkg/policy/policy.go:65]: i dont think this does what you think it does but also this wont work. i will fix it

File: pkg/rbac/rego.go (hunk 6 above)
Comment (emosbaugh) [reply pkg/rbac/rego.go:59]: default includes it

**Review by salahalsaleh — APPROVED** [review]

**Pull request merged**